### ISSUE TITLE
Mac M1 point cloud example freezes when trying to quit

### ISSUE THREAD

**Issue by jangkj09** [isl-org/Open3D#4813]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

This issue seems to be similar to this one: https://github.com/isl-org/Open3D/issues/3862

I installed within a conda enviroment with python 3.8 and was just trying to run the point cloud example: http://www.open3d.org/docs/release/tutorial/geometry/pointcloud.html

When it starts up, I get the following error: 
`[Open3D WARNING] GLFW Error: Cocoa: Failed to find service port for display`

Even so, a window shows up and the examples seems to run fine. When I try to quit by either closing the window or pressing 'ESC' or 'q', the window freezes and I have to force quit the kernel.

### Steps to reproduce the bug

```python
import open3d as o3d
import numpy as np

print("Load a ply point cloud, print it, and render it")
ply_point_cloud = o3d.data.PLYPointCloud()
pcd = o3d.io.read_point_cloud(ply_point_cloud.path)
print(pcd)
print(np.asarray(pcd.points))
o3d.visualization.draw_geometries([pcd],
                                  zoom=0.3412,
                                  front=[0.4257, -0.2125, -0.8795],
                                  lookat=[2.6172, 2.0475, 1.532],
                                  up=[-0.0694, -0.9768, 0.2024])
```

### Error message

Window closing just hangs.

### Expected behavior

Visualizer window closes cleanly.

### Open3D, Python and System information

```markdown
- Operating system: macOS 12.2.1
- Python version: Python 3.8.12 | packaged by conda-forge |
- Open3D version: output from python: 0.15.1
- System architecture: apple-silicon
- Is this a remote workstation?: no
- How did you install Open3D?: pip
- Compiler version (if built from source):
```

**Comment by yxlao** [comment]
Hi, it works fine for me. Did you run it inside Jupyter or directly with Python?
![Screen Recording 2022-02-28 at 18 22 59](https://user-images.githubusercontent.com/1501945/156093038-b9148bdd-d95d-4797-93b9-a25a552adb4b.gif)

**Comment by jangkj09** [comment]
Hi,  I tried both from the terminal from Python and also within Jupyter. It's not the warning that I am concerned about but everything hangs when I try to close the window and the window does not close. I have to force quit in order to close the window.

**Comment by yxlao** [comment]
@jangkj09 I have the exact OS version and Open3D version as you. As you can see from my recording, the window closes normally.

**Comment by jangkj09** [comment]
Ok. I'll just try it again on my end. There must be some quirk on my system I'm not aware of. It's a more general question, but is there something I can look into to better understand what's going on? Like, are there some log files created somewhere or some command to generate relevant logs?

**Comment by jangkj09** [comment]
Actually, I did what you did and added the example to a script file and executed the script. 
In this case, it did close properly and the script ended just like your recording. 

When I started the interpreter and input the example line-by-line, that's when closing the window would cause the interpreter to hang. A similar problem occurs when running from a jupyter notebook. I'm not sure what it is, but I will close this issue for now. Thanks!

**Comment by yxlao** [comment]
@jangkj09 Thanks, that's good to know. We'll test on the IPython side as well.

**Comment by seanreynoldscs** [comment]
Can you link to that other issue for Jupyter? I'm running into this crash when using Jupyter on my m1.

When I call o3d.visualization.draw_geometries([mesh]) it opens up as expected, but when I close the window it crashes python and I am forced to restart my jupyter instance from the terminal window.

**Comment by chinitaberrio** [comment]
Same issue here. Has this been solved?

**Comment by daniilpastukhov** [comment]
Facing same issue, is there any known solution?

**Comment by sszxc** [comment]
Facing same issue with jupyter notebook here, any progress?

**Comment by sszxc** [comment]
Facing same issue with jupyter notebook here, any progress?

**Comment by ChaosPredictor** [comment]
Similar here with Apple M2, on:
Python: 3.8.16
open3d: 0.17.0

**Comment by Khoa-NT** [comment]
> Similar here with Apple M2, on: Python: 3.8.16 open3d: 0.17.0
> 
> Please reopen the issue

Same case but with Python 3.9.12

**Comment by gghh121** [comment]
Same issue.
When I use jupyter-notebook, run this code-"o3d.visualization.draw_geometries([mesh])", the window will open normally, but when I press 'Q' or close the window, the window will be freeze, I must force quit.

Apple M2
Python: 3.10.14
open3d: 0.18.0
jupyter: 1.0.0

<img width="690" alt="image" src="https://github.com/isl-org/Open3D/assets/38463422/1132fcdd-be2a-469e-8c7a-440c18058fff">

**Comment by gegewen** [comment]
I am encountering the same issue. Window freeze when clicked close.

Apple M3 Max 
open3d: 0.18.0
notebook: 7.1.3

[Open3D WARNING] GLFW Error: Cocoa: Failed to find service port for display
2024-05-15 15:18:35.936 Python[21362:8808010] WARNING: Secure coding is automatically enabled for restorable state! However, not on all supported macOS versions of this application. Opt-in to secure coding explicitly by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState:.

**Comment by Thombela** [comment]
I fixed the issue, I think it might only work with meshes and not point cloud, I am not too sure even if it will work with all, but it worked with mine, I don't know if it will be a good enough fix for people, but for me it was a temporary fix that still didn't leave me satisfied.

To remove the issue of it freezing after played around with my code and found that the issue was this line of code:

`
mesh = o3d.io.read_triangle_mesh(mesh_path, True)
`

I found that this is what caused my terminal to freeze during the process of closing my visualiser. Now I changed it to:

`
mesh = o3d.io.read_triangle_mesh(mesh_path)
`

I just removed the 2nd parameter which is responsible for post-processing. Removing this playes a huge role as post-processing often improves the visual quality of rendered images by applying effects such as anti-aliasing, shadows, or tone mapping. In Open3D, this is likely affecting how the mesh is displayed, including lighting effects and smoothness of the object. With post-processing enabled, the mesh can look more realistic or refined by enhancing details like lighting and shading.

Here is how my mesh looked when it had post-processing:
![result](https://github.com/user-attachments/assets/9c192666-0c0f-4caa-b2ea-7d4ec1cbb54b)
Here is how it looks now:
![result](https://github.com/user-attachments/assets/271c0a8b-aef3-4d87-91df-64bfbae0358f)
